src/ex4/server/services/item_manager.js

```diff
@@ -87,41 +111,57 @@ class ItemManager {
         itemName: taskInput,
         status: isCompleted,
         doneAt: null,
+        position: position,
       };
-      this.tasks.push(task);
       return await this.saveTaskToDB(task);
     }
   }
 
-  async toggleCompleted(id) {
-    const task = this.tasks.find((task) => {
-      return task.id == id;
-    });
-
+  async updateTask(taskID, taskToUpdate) {
```

Comment:
this method can also be simplified, you can constract the "updates" object and call the Item.Update only once